apps/meteor/ee/server/configuration/saml.ts

```diff
@@ -21,7 +23,27 @@ onLicense('saml-enterprise', () => {
 				value = value.split(',');
 			}
 
-			userObject.roles = ensureArray<string>(value);
+			const savedRoles = await Roles.find(
+				{},
+				{
+					projection: {
+						_id: 1,
+						name: 1,
+					},
+				},
+			).toArray();
+
+			const roles = ensureArray<string>(value)
+				.map((role) => {
+					const savedRole = savedRoles.find((r) => r._id === role) ?? savedRoles.find((r) => r.name === role);
+
+					if (!savedRole) return '';
+
+					return savedRole.name;
+				})
+				.filter(Boolean);
```

Comment:
```suggestion
			const roles = ensureArray<string>(value)
				.filter((role) => {
					const savedRole = savedRoles.find((r) => r._id === role) ?? savedRoles.find((r) => r.name === role);
					return savedRole;
				});
```
I think you could do this with a single filter.

Reply:
This filter does not fix the purpose of the task, which is to add support for the user to input both the id or the name in the role field.